store/migration/manager.go

```diff
@@ -1,37 +1,87 @@
 package migration
 
 import (
+	"encoding/binary"
+	"fmt"
+	"sync"
+	"time"
+
 	"golang.org/x/sync/errgroup"
 
 	"cosmossdk.io/log"
 	"cosmossdk.io/store/v2"
+	"cosmossdk.io/store/v2/commitment"
 	"cosmossdk.io/store/v2/snapshots"
+	"cosmossdk.io/store/v2/storage"
 )
 
 const (
 	// defaultChannelBufferSize is the default buffer size for the migration stream.
 	defaultChannelBufferSize = 1024
 	// defaultStorageBufferSize is the default buffer size for the storage snapshotter.
 	defaultStorageBufferSize = 1024
+
+	migrateChangesetKeyFmt = "m/cs_%x" // m/<version>
 )
 
+// VersionedChangeset is a pair of version and Changeset.
+type VersionedChangeset struct {
+	Version   uint64
+	Changeset *store.Changeset
+}
+
 // Manager manages the migration of the whole state from store/v1 to store/v2.
 type Manager struct {
 	logger           log.Logger
 	snapshotsManager *snapshots.Manager
 
-	storageSnapshotter snapshots.StorageSnapshotter
-	commitSnapshotter  snapshots.CommitSnapshotter
+	stateStorage    *storage.StorageStore
+	stateCommitment *commitment.CommitStore
+
+	db              store.RawDB
+	migratedVersion uint64
+	mtx             sync.Mutex
```

Comment:
What's the mutex for? Generally speaking, the mtx should be above all fields it protects so it's clear to the reader.